firewood/src/merkle/proof.rs

```diff
@@ -563,26 +447,25 @@ fn locate_subproof(
             #[allow(clippy::indexing_slicing)]
             let data = n.chd_encode()[index]
                 .as_ref()
-                .ok_or(ProofError::InvalidData)?
-                .to_vec();
+                .ok_or(ProofError::InvalidData)?;
             generate_subproof(data).map(|subproof| (Some(subproof), key_nibbles))
         }
     }
 }
 
-fn generate_subproof(encoded: Vec<u8>) -> Result<SubProof, ProofError> {
+fn generate_subproof_hash(encoded: &[u8]) -> Result<[u8; 32], ProofError> {
```

Comment:
nit: Proof code uses the type TrieHash for this. Can we reuse it?

Reply:
Will do this in a follow up